crates/biome_parser/CONTRIBUTING.md

```diff
@@ -0,0 +1,382 @@
+# Parser
+
+This document will explain how to contribute to creating a new Parser with Biome.
+
+## Grammar
+
+Everything starts with writing the grammar for the language. We use a fork of [`ungrammar`](https://github.com/rust-analyzer/ungrammar), a simple DSL
+for writing grammar for languages.
+
+Internally, we have a codegen script that reads `.ungram` files and all the files needed to write a parser.
+
+### Create a new language
+
+From the root of this repository, head to `xtask/codegen` folder. That's where you'll start doing your work.
+
+Let's say you want to create a new language that has the extension `html`, you'll have to follow these instructions:
+
+1. Create a new `html.ungram` file in this folder.
+   Add this legend to the `.ungram` file.
+
+  ```
+  // This grammar specifies the structure of Rust's concrete syntax tree.
+  // It does not specify parsing rules (ambiguities, precedence, etc are out of scope).
+  // Tokens are processed -- contextual keywords are recognised, compound operators glued.
+  //
+  // Legend:
+  //
+  //   //                    -- comment
+  //   Name =                -- non-terminal definition
+  //   'ident'               -- token (terminal)
+  //   A B                   -- sequence
+  //   A | B                 -- alternation
+  //   A*                    -- zero or more repetition
+  //   (A (',' A)* ','?)            -- repetition of node A separated by ',' and allowing a trailing comma
+  //   (A (',' A)*)             -- repetition of node A separated by ',' without a trailing comma
+  //   A?                    -- zero or one repetition
+  //   (A)                   -- same as A
+  //   label:A               -- suggested name for field of AST node
+  ```
+1. Create a new file called `src/html_kinds_src.rs`. This file must return a static `KindSrc`.
+1. Create two new crates: `biome_html_syntax` and `biome_html_factory`. Use `cargo new --lib crates/biome_html_syntax`.
+1. Create a `generated/` folder inside the `src/` folder of the newly created crates.
+1. Add a new variant to `LanguageKind`, inside `language_kind.rs` file. The new variant will be `Html`. You'll have to implement
+   all methods and cover the new variant.
+1. Add a new prefix `html_` to `LANGUAGE_PREFIXES` inside `language_kind.rs`.
+1. Once you covered all variants, run the command `cargo codegen grammar`.
```

Comment:
list item number should be increasing

Reply:
That's the trick, if you leave them to all `1` they will be automatically **rendered** with increased numbers :) 

This means that if you need to add a new item to the list, you don't need to update all the numbers. Check the [rendered document](https://github.com/biomejs/biome/blob/docs/better-parser-contribution-guide/crates/biome_parser/CONTRIBUTING.md#create-a-new-language)

Reply:
🤯 I just learned new tricks.